drizzle-orm/src/mysql-core/dialect.ts

```diff
@@ -334,6 +335,54 @@ export class MySqlDialect {
 		return sql`${withSql}select${distinctSql} ${selection} from ${tableSql}${joinsSql}${whereSql}${groupBySql}${havingSql}${orderBySql}${limitSql}${offsetSql}${lockingClausesSql}`;
 	}
 
+	buildSetOperationQuery({
+		operator,
+		isAll,
+		leftSelect,
+		rightSelect,
+		limit,
+		orderBy,
+		offset,
+	}: MySqlSetOperatorConfig): SQL {
+		const leftChunk = sql`(${leftSelect.getSQL()}) `;
+		const rightChunk = sql`(${rightSelect.getSQL()})`;
+
+		let orderBySql;
+		if (orderBy && orderBy.length > 0) {
+			const orderByValues: SQL<unknown>[] = [];
+
+			// The next bit is necessary because the sql operator replaces ${table.column} with `table`.`column`
+			// which is invalid MySql syntax, Table from one of the SELECTs cannot be used in global ORDER clause
+			for (const orderByUnit of orderBy) {
+				if (is(orderByUnit, MySqlColumn)) {
+					orderByValues.push(sql.raw(orderByUnit.name));
```

Comment:
You probably need this value escaped in the query?
(also check that all other places where you insert the table/column name directly use `sql.identifier`)
```suggestion
					orderByValues.push(sql.identifier(orderByUnit.name));
```